### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jmford-engr**

### CODE REVIEW

**Comment by jbrzozoski** [comment]
You did something weird with the commits...  You have the same commit twice on two different branches then merged into each other.  Commits e3b00dc and 432f551 have identical file contents (I double-checked).  But it looks like commit e3b00dc is the one you want to keep (that should pass ECA).  Github generates the merge back to develop for you when the pull request is completed.  So, I think what you want to do this in your local git workspace:
1. Checkout the develop branch that has all of your changes (e.g. `git checkout develop`)
2. Do a hard reset back to 83b00dc (e.g. `git reset --hard 83b00dc`)
3. Do a force push back into your fork of the repository (e.g. `git push myfork --force`)

I believe that should auto-update this pull request with the right information.

I'm still reviewing the code as well.  I may have some comments on changes to make there as well if you want to wait.

File: sparkplug_b/stand_alone_examples/c/template_as_custom_props/example.c
Comment (jbrzozoski) [inline sparkplug_b/stand_alone_examples/c/template_as_custom_props/example.c:212]: The functionality this `seq` variable provided has been replaced by the function `reset_sparkplug_sequence`.  You can either put a call to that function here, or (preferably) down around line 228 just before the call to `get_next_payload` when building the NBIRTH message.

File: sparkplug_b/stand_alone_examples/c/udt_example/example.c
Comment (jbrzozoski) [inline sparkplug_b/stand_alone_examples/c/udt_example/example.c:347]: Also needs a call to `reset_sparkplug_sequence` put in to replace this, preferably around line 363 just before the call to `get_next_payload` for the NBIRTH.

File: sparkplug_b/stand_alone_examples/c/udt_example/example.c
Comment (jbrzozoski) [inline sparkplug_b/stand_alone_examples/c/udt_example/example.c:488]: Good catch on noticing this was missing.

**Review by jbrzozoski — CHANGES_REQUESTED** [review]
I noticed a few things during my read through.

Also, if you want, every occurrence of the form `new= malloc(strlen(blah)+1)` followed `strcpy(new)` can just be replaced with `new = strdup(blah)`.  Makes lots of code easier to read.

**Comment by jmford-engr** [comment]
On Thu, Nov 5, 2020 at 7:46 PM jbrzozoski <notifications@github.com> wrote:

> *@jbrzozoski* requested changes on this pull request.
>
> I noticed a few things during my read through.
>
> Also, if you want, every occurrence of the form new=
> malloc(strlen(blah)+1) followed strcpy(new) can just be replaced with new
> = strdup(blah). Makes lots of code easier to read.
>
Agreed.  Done.

> ------------------------------
>
> In sparkplug_b/stand_alone_examples/c/template_as_custom_props/example.c
> <https://github.com/eclipse/tahu/pull/106#discussion_r518489852>:
>
> > @@ -209,7 +209,6 @@ void my_log_callback(struct mosquitto *mosq, void *userdata, int level, const ch
>  void publish_births(struct mosquitto *mosq) {
>  	// Initialize the sequence number for Sparkplug MQTT messages
>  	// This must be zero on every NBIRTH publish
> -	seq = 0;
>
> The functionality this seq variable provided has been replaced by the
> function reset_sparkplug_sequence. You can either put a call to that
> function here, or (preferably) down around line 228 just before the call to
> get_next_payload when building the NBIRTH message.
> ------------------------------
>
> In sparkplug_b/stand_alone_examples/c/udt_example/example.c
> <https://github.com/eclipse/tahu/pull/106#discussion_r518490452>:
>
> > @@ -344,7 +344,6 @@ void my_log_callback(struct mosquitto *mosq, void *userdata, int level, const ch
>  void publish_births(struct mosquitto *mosq) {
>  	// Initialize the sequence number for Sparkplug MQTT messages
>  	// This must be zero on every NBIRTH publish
> -	seq = 0;
>
> Also needs a call to reset_sparkplug_sequence put in to replace this,
> preferably around line 363 just before the call to get_next_payload for
> the NBIRTH.
> ------------------------------
>
> In sparkplug_b/stand_alone_examples/c/udt_example/example.c
> <https://github.com/eclipse/tahu/pull/106#discussion_r518491239>:
>
> >
>          // Publish the NBIRTH on the appropriate topic
>          mosquitto_publish(mosq, NULL, "spBv1.0/Sparkplug B Devices/NBIRTH/C Edge Node 1", message_length, binary_buffer, 0, false);
>
>  	// Free the memory
>  	free(binary_buffer);
> -	free(nbirth_payload.uuid);
> +	free(row_data);
>
> Good catch on noticing this was missing.
>

Credit valgrind for the find!

**Comment by jmford-engr** [comment]
Not sure what to do from here.  I tried your suggestions but it still says I'm not covered.  Should we trash this PR and start a new one?

**Comment by jbrzozoski** [comment]
The code changes all look good.  Looks like you forgot the "Signed-off-by:" footer in your commit messages, though.
I made a cleaned up version of your commits over here:
https://github.com/jbrzozoski/tahu/tree/pull_106
The commits all still have your name and sign-off on them.  If you like the way that all looks, you can either imitate that change on your develop branch, or we can abandon this pull request and make another one from my branch.  Let me know which way you'd rather go.

**Comment by jmford-engr** [comment]
Looks good.  Can you just abandon mine and make a new one from your
branch?  I'll hopefully get it right next time!

**Comment by jbrzozoski** [comment]
I just opened the pull request for the cleaned up version as #107 
I don't have permissions to close this pull request.  I believe only you can (as the author) or one of the official maintainers can.

**Pull request closed without merging**
